tests/ui/type_param_mismatch.stderr

```diff
@@ -0,0 +1,64 @@
+error: this type parameter is named B, but it does not refer to Foo::B
```

Comment:
I find the error message to be confusing. It's not clear what `does not refer to Foo::B` is about.

I think the clarity can be improved. Perhaps something like this?

```suggestion
error: `Foo` has a similarly named generic type parameter `B` in its declaration, but in a different order.
```

Reply:
Thanks for the suggestion, that sounds better.